src/mlpack/methods/ann/layer/atrous_convolution_impl.hpp

```diff
@@ -37,6 +37,65 @@ AtrousConvolution<
   // Nothing to do here.
 }
 
+template<
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename InputDataType,
+    typename OutputDataType
+>
+AtrousConvolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+>::AtrousConvolution(
+  const AtrousConvolution& network) :
+    inSize(network.inSize),
+    outSize(network.outSize),
+    kernelWidth(network.kernelWidth),
+    kernelHeight(network.kernelHeight),
+    strideWidth(network.strideWidth),
+    strideHeight(network.strideHeight),
+    inputWidth(network.inputWidth),
+    inputHeight(network.inputHeight),
+    outputWidth(network.outputWidth),
+    outputHeight(network.outputHeight),
+    dilationWidth(network.dilationWidth),
+    dilationHeight(network.dilationHeight),
+    weight(network.weight),
+    bias(network.bias),
+    paddingType(network.paddingType),
+    padH(network.padH),
+    padW(network.padW)
```

Comment:
For these parameters, it seems like it would just be better to call `padding(network.padding)` to copy the internally-held padding layer.  Is there any reason we can't do that?  (Maybe I overlooked something.)

Reply:
I will try this.